testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/priority/resource/ProviderPriorityResource.java

```diff
@@ -32,7 +32,7 @@ public String paramConverter(@PathParam("foo") ProviderPriorityFoo foo) throws E
    @GET
    @Path("register")
    public Response register() throws Exception {
-      ResteasyProviderFactory factory = ResteasyProviderFactory.getInstance();
+      ResteasyProviderFactory factory = ResteasyProviderFactory.newInstance();
```

Comment:
I have some doubts about the change here. In this case, the attempt is exactly to alter the current ResteasyProviderFactory instance (yes, I agree, that's not a best practice at all). By creating a new instance and not setting it in the singleton reference within ResteasyProviderFactory, you're altering the test. I suspect this is not spotted by Travis CI because the test here is likely not running due to the exclusions for RESTEASY-1935.

Reply:
Do you think you can let the test do what it currently does and cleanup/restore the initial status afterwards?